testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/AbstractKeycloakTest.java

```diff
@@ -638,4 +640,65 @@ protected static InputStream httpsAwareConfigurationStream(InputStream input) th
         }
         return in;
     }
+
+    /**
+     * This static method can be used to change default {@ode WebDriver} to different one if the test is able to run
+     * with only one {@code WebDriver}.
+     *
+     * {@link target} parameter can be of type {@link AbstractKeycloakTest} or {@link AbstractPage}.
+     *
+     * @param target target class on which one needs to change {@code WebDriver}
+     * @param driver the new {@code WebDriver}
+     */
+    public void replaceDefaultWebDriver(Object target, WebDriver driver) {
+        DroneUtils.addWebDriver(driver);
+        List<Field> allFields = new ArrayList<>();
+        // Add all fields of this class and superclasses
+        Class<?> targetClass = target.getClass();
+        while (targetClass != null) {
+            allFields.addAll(Arrays.asList(targetClass.getDeclaredFields()));
+            allFields.addAll(Arrays.asList(targetClass.getFields()));
+            targetClass = targetClass.getSuperclass();
+        }
+
+        for (Field f : allFields) {
+            if (f.getAnnotation(Page.class) != null
+                    && AbstractPage.class.isAssignableFrom(f.getType())) {
+                try {
+                    if (!f.isAccessible())
+                        f.setAccessible(true);
+                    Object o = f.get(target);
+                    AbstractPage page = (AbstractPage) o;
+                    page.setDriver(driver);
+                    replaceDefaultWebDriver(page, driver);
+                } catch (IllegalAccessException e) {
+                    throw new IllegalStateException("Could not replace the driver in " + f, e);
+                }
+            } else if (AbstractKeycloakTest.class.isAssignableFrom(target.getClass())
+                    && f.getName().equals("driver")) {
+                try {
+                    if (!f.isAccessible())
+                        f.setAccessible(true);
+                    f.set(target, driver);
+
+                } catch (IllegalAccessException e) {
+                    throw new IllegalStateException("Could not replace the " + f.getName() + " in class "
+                            + target.getClass().getName(), e);
+                }
+            } else if (AbstractKeycloakTest.class.isAssignableFrom(target.getClass())
+                    && f.getName().equals("oauth")) {
+                try {
+                    if (!f.isAccessible())
+                        f.setAccessible(true);
+                    Object o = f.get(target);
+                    ((OAuthClient) o).setDriver(driver);
+                } catch (IllegalAccessException e) {
+                    throw new IllegalStateException("Could not replace the " + f.getName() + " in class "
+                            + target.getClass().getName(), e);
+                }
+            }
+        }
+        DroneUtils.addWebDriver(driver);
```

Comment:
There is a call "DroneUtils.addWebDriver" at both the beginning and end of this method. Is it intentional or a typo?

Reply:
you are right, fixed to removeWebDriver()